entry.go

```diff
@@ -261,7 +261,15 @@ func (entry *Entry) log(level Level, msg string) {
 }
 
 func (entry *Entry) fireHooks() {
-	err := entry.Logger.Hooks.Fire(entry.Level, entry)
+	var tmpHooks LevelHooks
+	entry.Logger.mu.Lock()
+	tmpHooks = make(LevelHooks, len(entry.Logger.Hooks))
+	for k, v := range entry.Logger.Hooks {
+		tmpHooks[k] = v
+	}
+	entry.Logger.mu.Unlock()
```

Comment:
This could be simplified:
```go
entry.Logger.mu.Lock()
tmpHooks := append(LevelHooks(nil), entry.Logger.Hooks...)
entry.Logger.mu.Unlock()
```

Reply:
it may lead to more reallocation and copy depending on the array size though.

Reply:
or not

Reply:
🤷‍♂️